test/src/requestinterception.spec.ts

```diff
@@ -192,11 +199,13 @@ describe('request interception', function () {
 
       await page.setExtraHTTPHeaders({referer: server.EMPTY_PAGE});
       await page.setRequestInterception(true);
-      page.on('request', request => {
-        expect(request.headers()['referer']).toBe(server.EMPTY_PAGE);
+      let request!: HTTPRequest;
+      page.on('request', req => {
+        request = req;
```

Comment:
I think it is important to check that the data is already there at the time of the callback. Checking later might hide the fact that the data is not available when the user expects it,

Reply:
Here is an example. And this passes:
```
   it.only('should intercept', async () => {
      const {page, server} = await getTestState();

      await page.setRequestInterception(true);
      let request!: HTTPRequest;
      page.on('request', req => {
        if (isFavicon(req)) {
          void req.continue();
          return;
        }
        request = req;
        void req.continue();
        expect(false).toBeTruthy();
      });
      const response = (await page.goto(server.EMPTY_PAGE))!;

      expect(request).toBeTruthy();
      expect(response.ok()).toBe(true);
      expect(response.remoteAddress().port).toBe(server.PORT);
```
I think we can wrap the error, and throw if it not `undefined`?

Reply:
I meant that request headers should be asserted in the event callback and not after the whole navigation is done

Reply:
I understand what you are saying. But `expect` in the call back has no effect on the passing of the test. 
I have `expect(false).toBeTruthy();` in my example an the test still Passes.

Reply:
Why does it have no effect? are we catching errors from events somewhere?

Reply:
Went though our implementation and `mitt` it never catches the error not sure why its not reported.

Reply:
I think it is some mocha quirk, running standalone reports an unhandled promise rejection. Also, does Node not crash on that anymore?

Reply:
For node it becomes  - `unhandled-rejections` and we would need to set this in `.mocharc.js`:
```
'node-option': ['unhandled-rejections=strict'],
```

Reply:
Let's get the data during the event but assert it after the event? it would be easier than catching errors

Reply:
How about we wrap the expects in a `try/catch` and store it in an `error` then if its not empty throw?

Reply:
Done + added EsLint Rule